README.md

```diff
@@ -1,150 +1,197 @@
-
-```
+Each major version can be built independently.
+```bash
 nix build .#psql_14/bin -o result-14
 nix build .#psql_15/bin -o result-15
 ```
 
-This will create two symlinks named `result-14` (resp. `result-15`). You can use the resulting symlinks as a binary distribution; as if they were unpacked tarballs.
+Builds Postgres 14 & 15 with all of our extensions and symlinks the resulting build artifaces e.g. `psql`, `pg_dump` etc at `result-15` and `result-15` respectively.
 
-### PostgreSQL Docker Images
 
-From the binary images generated by `nix build`, we also create Docker images. You can create
-and load an image like so:
+On first run these commands will take a long time to complete. Intermediate results are cached and reused leading to significantly reduced built times on subsequent runs.
 
-```
-nix build .#psql_15/docker -o result-15
-docker load -i ./result-15
-```
+The Postgres build itself is defined upstream in [nixpkgs#postgresqlXX](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/databases/postgresql.nix). That distribution is frequently updated and historically has released minor and major updates within 10 days. We can optionally override the version number or vendor the derivation to support additional versions.
+
+Note that nixpkgs supports 1 minor version (the most recent) for each major version.
 
-There will then be an image named `postgresql-15:latest` tagged in local image
-registry.
 
-This repository also provides pre-packaged docker images hosted via GitHub Packages; simply download them with:
+To launch a psql repl with the built version, first start the server:
 
+```bash
+nix run .#start-server 15
 ```
-docker pull ghcr.io/supabase/nix-postgres-14:latest
-docker pull ghcr.io/supabase/nix-postgres-15:latest
+Next, launch the psql client.
+
+```bash
+nix run .#start-client 15
 ```
 
-### A full suite of extensions
+Both commands are defined in [flake.nix](./flake.nix)
+
+## Building multiple versions of an extension: e.g. pgvector
 
-The binary distributions have many extensions enabled; these include:
+Extensions included with the Postgres distribution come from two sources:
 
-- postgis
-- pgrouting
-- pgtap
-- pg_cron
-- pgaudit
-- pgjwt
-- plpgsql_check
-- pg_safeupdate
-- timescaledb
-- wal2json
-- plv8
-- rum
-- pgvector
-- pg_repack
-- pgroonga
-- pgsql
-- pg_plan_filter
-- pg_net
-- pg_hashids
-- pgsodium
-- pg_graphql
-- pg_stat_monitor
-- pg_jsonschema
-- vault
-- hypopg
-- pg_tle
-- wrappers
-- supautils
+- Upstream in [nixpkgs](https://github.com/NixOS/nixpkgs/tree/a996f7185eaaa8ae261adbc6b772761d62869796/pkgs/servers/sql/postgresql/ext)
+- Locally in [./ext](./ext)
 
-You can just use `CREATE EXTENSION` to enable most of these. Some may require
-tweaks to [postgresql.conf](./tests/postgresql.conf) to enable.
+Each extension has a default version that can be easily updated or overriden, but maintaining multiple versions of each extension long-term is not practical with current tooling.
```

Comment:
Hm, this sounds like we cannot have builds running with different versions of an extension.

As Austin mentioned, we can do that by using different git tags or branches. i.e. we can have `git tag postgresql-15-with-supautils-1.1` or `git branch potgresql-15-with-supautils-1.2`

Maybe that could be clarified.